File: fintechplatform/src/main/java/com/fintechplatform/android/card/db/PaymentCardPersistanceModule.java
package com.fintechplatform.android.card.db;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PaymentCardPersistanceModule {

    @Provides
    @Singleton
    PaymentCardPersistenceDB providesPayamentCardPersistance(PaymentCardDB paymentCardDb) {
        return new PaymentCardPersistenceDB(paymentCardDb);
    }

    @Provides
    @Singleton
    PaymentCardDB providesPaymentCardDB(){
        return new PaymentCardDB();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/card/db/PaymentCardPersistanceModule.kt
package com.fintechplatform.ui.card.db

import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class PaymentCardPersistanceModule {

    @Provides
    @Singleton
    internal fun providesPayamentCardPersistance(paymentCardDb: PaymentCardDB): PaymentCardPersistenceDB {
        return PaymentCardPersistenceDB(paymentCardDb)
    }

    @Provides
    @Singleton
    internal fun providesPaymentCardDB(): PaymentCardDB {
        return PaymentCardDB()
    }
}